src/storage/store/chunked_node_group.cpp

```diff
@@ -461,5 +464,36 @@ std::unique_ptr<ChunkedNodeGroup> ChunkedNodeGroup::deserialize(MemoryManager& m
     return chunkedGroup;
 }
 
+void ChunkedNodeGroup::setUnused(MemoryManager& mm) {
+    dataInUse = false;
+    mm.getBufferManager()->getSpillerOrSkip([&](auto& spiller) { spiller.addUnusedChunk(this); });
+}
+
+void ChunkedNodeGroup::loadFromDisk(MemoryManager& mm) {
+    mm.getBufferManager()->getSpillerOrSkip([&](auto& spiller) {
+        std::unique_lock lock{spillToDiskMutex};
```

Comment:
Would it be better to assert on `dataInUse` here?

Reply:
No, if the chunk hasn't already been spilled to disk it's possible for it to be spilled to disk as this is running, if another thread needs the memory.
`dataInUse` could actually be true here since the last chunked group in each collection won't ever get marked as unused.